src/main/java/com/google/gfx/services/matching/MatchingService.java

```diff
@@ -0,0 +1,38 @@
+package com.google.gfx.services.matching;
+
+import com.google.gfx.model.Match;
+import com.google.gfx.model.Order;
+
+/**
+ * Matches existing orders with other existing or new ones.
+ */
+public interface MatchingService {
+
+  /**
+   * This method was conceived to be ran as part of a cron job that matches existing buy and sell
+   * orders.
+   */
+  void matchExistingBuyAndSellOrders();
+
+  /**
+   * Creates a match for an order, that must be accepted by the order owner.
+   *
+   * @param order original posted order
+   * @param amount amount of the match, which might fully or partially match the original order
+   */
+  void matchOrder(Order order, float amount);
+
+  /**
+   * Accepts a proposed match.
+   *
+   * @param match proposed match
+   */
+  void acceptMatch(Match match);
```

Comment:
One thing that jumped to sight was that some of these methods should probably belong to the objects. For instance, acceptMatch() should probably be Match.accept(), instead. Or they could both exist simultaneously, e.g., MatchingService controls whether a match can be decided on by a user and then Match.accept() just registers the decision (or even Match.setBuyerResponse()/Match.setSellerResponse()).

I can't make my mind up on what's best: whether to check if a user can decide on a match inside the acceptMatch() method or outside..